File: 094-inorder-traversal/solution.js
const TreeNode = require('../datastructure/Tree/TreeNode');

/**
 * @param {TreeNode} root
 * @return {number[]}
 */
let num = []
var inorderTraversal = function(root) {
    if (root === null)
        return num
    inorderTraversal(root.left)
    num.push(root.val)
    inorderTraversal(root.right)
    return num
};

File: 094-inorder-traversal/solution.py
from typing import List
from typing import Optional
from Tree import TreeNode
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        num = []
        def dfs(node: Optional[TreeNode]):
            if node is None:
                return
            dfs(node.left)
            num.append(node.val)
            dfs(node.right)
        dfs(root)
        return num